components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/AuthUtil.java

```diff
@@ -0,0 +1,169 @@
+/*
+ *  Copyright WSO2 Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.apimgt.gateway.handlers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.apimgt.gateway.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.APIManagerConfiguration;
+import org.wso2.carbon.apimgt.impl.dto.APIKeyValidationInfoDTO;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import javax.cache.Cache;
+import javax.cache.CacheConfiguration;
+import javax.cache.Caching;
+import java.util.concurrent.TimeUnit;
+
+
+public class AuthUtil {
+    private static Logger log = LoggerFactory.getLogger(AuthUtil.class);
+
+    public AuthUtil(){
+
+        initParams();
+
+        this.getGatewayKeyCache();
+
+    }
+
+    private boolean removeOAuthHeadersFromOutMessage=true;
+
+    protected void initParams() {
+        try {
+            APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfiguration();
+            String cacheEnabled = config.getFirstProperty(APIConstants.GATEWAY_TOKEN_CACHE_ENABLED);
+            if(cacheEnabled != null)
+            gatewayTokenCacheEnabled = Boolean.parseBoolean(cacheEnabled);
+            String value = config.getFirstProperty(APIConstants.REMOVE_OAUTH_HEADERS_FROM_MESSAGE);
+            if (value != null)
+                removeOAuthHeadersFromOutMessage = Boolean.parseBoolean(value);
+        } catch (Exception e) {
+            log.error("Did not found valid API Validation Information cache configuration. Use default configuration" + e);
```

Comment:
log with exception as (message ,e)

Avoid catching generic exceptions. The exception not thrown upwards any reason for that?